modules/internal/String.chpl

```diff
@@ -961,8 +1171,31 @@ module String {
 
         // Edge cases
         if count {
-          if nLen == 0 then
-            localRet = thisLen+1;
+          if nLen == 0 { // Empty needle
+            const isByteIndexed =
+              if region.hasLowBound() then region.low.type == byteIndex else
+              if region.hasHighBound() then region.high.type == byteIndex
+              else false;
```

Comment:
I think it'd be more typical to write it as boolean operators inside the conditional on the next line... this is an interesting alternative though.